### ISSUE TITLE
Add a new `ParallelStrategy` that doesn't require gradient synchronization

### ISSUE THREAD

**Issue by ordabayevy** [cellarium-ai/cellarium-ml#175]
At the moment we use `DDPStrategy` to train models that don't require gradient synchronization (`IncrementalPCA`, `OnePassMeanVarStd`). However, ddp strategy requires that your model has parameters which is why we have `_dummy_param` in these models. This works but it get's a bit inconvenient when the trained model is used as a transform because it becomes and unused parameter and ddp complains about it.

**Comment by ordabayevy** [comment]
Starting from PyTorch Lightning 2.3 returning None in a training step is not allowed https://github.com/Lightning-AI/pytorch-lightning/pull/19918
